setup-extension/action.yml

```diff
@@ -114,6 +117,7 @@ runs:
       if: ${{ ! inputs.extension-zip }}
       uses: actions/checkout@v4
       with:
+        ref: ${{ inputs.extensionRef || github.ref }}
```

Comment:
when you need this?

Reply:
Currently, in a workflow which is supposed to trigger jobs with multiple platform+extension version combinations.

Of course, we could write a `.github/workflows/ci.yml` varying per branch, but I think it's easier maintenance-wise to have the workflow in one place and just check out current/legacy code depending on the test case.